tck/src/main/java/org/eclipse/sparkplug/tck/test/edge/ReceiveCommandTest.java

```diff
@@ -14,314 +14,300 @@
-					logger.info("Device birth received for " + levels[4]);
-					dbirth = true;
-				}
-			} else if (levels[2].equals("NDATA")) {
-				if (levels[0].equals(NAMESPACE) && levels[1].equals(group_id) && levels[3].equals(edge_node_id)) {
-					testResults.put("operational-behavior-data-commands-rebirth-action-1", FAIL);
-					logger.info("Error: Data received for edge node" + levels[3]);
-				}
-			} else if (levels[2].equals("DDATA")) {
-				if (levels[0].equals(NAMESPACE) && levels[1].equals(group_id) && levels[3].equals(edge_node_id)) {
-					testResults.put("operational-behavior-data-commands-rebirth-action-1", FAIL);
-					logger.info("Error: Data received for edge node" + levels[3] + " device id " + levels[4]);
-				}
-			}
-
-			if (state.equals("SendingNodeRebirth") && nbirth && dbirth) {
-				testResults.put("operational-behavior-data-commands-rebirth-action-2", PASS);
-				state = "DisconnectingClient";
-				nbirth = false;
-				dbirth = false;
-				disconnectClient(clientId);
-			}
-			if (state.equals("DisconnectingClient") && nbirth && dbirth) {
-				endTest();
-			}
-		}
-	}
-
-}
\ No newline at end of file
+    private static final String BD_SEQ = "bdSeq";
+    private static final String NODE_CONTROL_REBIRTH = "Node Control/Rebirth";
+    private static Logger logger = LoggerFactory.getLogger("Sparkplug");
+    private final @NotNull TCK theTCK;
+    private final @NotNull Map<String, String> testResults = new HashMap<>();
+    private final @NotNull ArrayList<String> testIds = new ArrayList<>();
+    private @NotNull status state;
+    private @NotNull String deviceId;
+    private @NotNull String groupId;
+    private @NotNull String edgeNodeId;
+    private @NotNull String hostApplicationId;
+    private @NotNull long deathBdSeq;
+    private String edgeNodeClientId;
+    private boolean bNBirth = false, bDBirth = false;
+    private PublishService publishService = Services.publishService();
+
+    public ReceiveCommandTest(TCK aTCK, String[] parms) {
+        logger.info("Edge Node payload validation test. Parameters: {} ", Arrays.asList(parms));
+        theTCK = aTCK;
+
+        if (parms.length < 3) {
+            logger.error("Parameters to edge receive command test must be: groupId edgeNodeId deviceId");
+            return;
+        }
+        state = status.START;
+        deathBdSeq = -1;
+        groupId = parms[0];
+        edgeNodeId = parms[1];
+        deviceId = parms[2];
+        logger.info("Parameters are  GroupId: {}, EdgeNodeId: {}, DeviceId: {}", groupId, edgeNodeId, deviceId);
+
+        testIds.add(ID_OPERATIONAL_BEHAVIOR_DATA_COMMANDS_REBIRTH_ACTION_1);
+        testIds.add(ID_OPERATIONAL_BEHAVIOR_DATA_COMMANDS_REBIRTH_ACTION_2);
+        testIds.add(ID_OPERATIONAL_BEHAVIOR_DATA_COMMANDS_REBIRTH_ACTION_3);
+
+        sendCommand(true);
+
+        // indicate we are testing the receipt of NBIRTH and DBIRTH messages after a rebirth command
+        // Dont understand this setting -- ???
```

Comment:
This just means that we set this assertion value to false by default, then track the receipt of both NBIRTH and DBIRTH messages for device attached to an edge node.